express-post-json/index.js

```diff
@@ -0,0 +1,27 @@
+const express = require('express');
+const app = express();
+
+let nextId = 1;
+const grades = {};
+
+app.get('/api/grades', (req, res) => {
+  const arr = [];
+  for (const property in grades) {
+    arr.push(grades[property]);
+  }
+  res.json(arr);
+});
+
+app.use(express.json());
+
+app.post('/api/grades', (req, res) => {
+  req.body.id = nextId;
+  res.status(201).send(req.body);
+  grades[1 - nextId] = req.body;
+  nextId++;
+});
```

Comment:
The response shouldn't be sent until after everything is finished being updated, otherwise the client may get a successful response even though part of the operation failed.